xarray/backends/netCDF4_.py

```diff
@@ -210,9 +216,9 @@ def __init__(self, filename, mode='r', format='NETCDF4', group=None,
         self.ds = opener()
```

Comment:
So I can catch `autoclose`, but am I trying to throw an error in `__getstate__` and `__setstate__` for pickle? Not sure where to go for that part of the problem.